pkg/controllers/controller.go

```diff
@@ -49,22 +49,24 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	}
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
-	// 3. Set to true to remove race condition
+	// 3. Set Defaults
+	if _, ok := resource.(apis.Defaultable); ok {
+		resource.(apis.Defaultable).SetDefaults(ctx)
+	}
+	// 4. Set to true to remove race condition
```

Comment:
This comment is confusing - what race condition?

Reply:
If we have multiple controllers reconciling a provisioner, they can set the status of the provisioner to different statuses, where last write wins. We're setting this to true for now while we change what resources the controllers are reconciling. I'll add this in too to make it clearer.